File: esbmc.cpp/validador_categoria/validador_espacos.cpp
#define MAX_LEN 31

using namespace std;

bool validate_no_spaces(const char category[MAX_LEN], int len) {
    for (int i = 0; i < len; i++) {
        __ESBMC_assert(category[i] != ' ', "A categoria não deve conter espaços.");
    }
    return true;
}

int main() {
    char category[MAX_LEN];
    int len;

    __ESBMC_assume(len >= 3 && len <= 30);

    validate_no_spaces(category, len);

    return 0;
}

File: esbmc.py/Testes02/validador_categoria/validador_espacos.py
MAX_LEN: int = 31

def validate_no_spaces(category: str, length: int) -> bool:
    i: int = 0
    while i < length:
        assert category[i] != ' ', "A categoria não deve conter espaços."
        i += 1
    return True


def main() -> None:
    category: str = input()      
    length: int = len(category)  

    assert length >= 3 and length <= 30

    validate_no_spaces(category, length)


main()
